File: src/main/java/io/dustin/finstage/financial/dto/ExternalSymbolListResponse.java
package io.dustin.finstage.financial.dto;

import lombok.Getter;

import java.util.List;

@Getter
public class ExternalSymbolListResponse {
    private int total;
    private int page;
    private int size;
    private int totalPages;
    private boolean hasNext;
    private boolean hasPrev;
    private List<ExternalSymbolItem> items;

    @Getter
    public static class ExternalSymbolItem {
        private String symbol;
        private String name;
        private String country;
    }
}

File: src/main/kotlin/io/dustin/finstage/financial/dto/ExternalSymbolListResponse.kt
package io.dustin.finstage.financial.dto

import com.fasterxml.jackson.annotation.JsonProperty

data class ExternalSymbolListResponse(
        @JsonProperty("total")
        val total: Int,

        @JsonProperty("page")
        val page: Int,

        @JsonProperty("size")
        val size: Int,

        @JsonProperty("totalPages")
        val totalPages: Int,

        @JsonProperty("hasNext")
        val hasNext: Boolean,

        @JsonProperty("hasPrev")
        val hasPrev: Boolean,

        @JsonProperty("items")
        val items: List<ExternalSymbolItem>
)

data class ExternalSymbolItem(
        @JsonProperty("symbol")
        val symbol: String,

        @JsonProperty("name")
        val name: String,

        @JsonProperty("country")
        val country: String
)
